shared/js/l10n.js

```diff
@@ -53,7 +53,12 @@
       try {
         xhr.send(null);
       } catch (e) {
-        callback(new L10nError('Not found: ' + url));
+        if (e.name === 'NS_ERROR_FILE_NOT_FOUND') {
+          // the app: protocol throws on 404, see https://bugzil.la/827243
+          callback(new L10nError('Not found: ' + url));
+        } else {
+          throw e;
+        }
```

Comment:
Yes :)  You helped me debug this, actually.  Parsing, `addAST` and emitting of the `dupliacateerror` event happens in the callback the the XHR's `load` event. `xhr.onload` is [called synchronously](https://github.com/mozilla-b2g/gaia/blob/d3d20d982308484951038c74cfd5eea6bb6a0d5a/build/webapp-optimize.js#L585-L599) from inside of `xhr.send` in the build system.  I want to throw when the buildtime bindings intercept `dupliacateerror` but because it all happens in `xhr.send`, the `try…catch` here acts like a catch-all for all thrown errors.  Hence the need to re-throw if it's not an error strictly related to fetching of resources.